src/services/ioSignService.ts

```diff
@@ -117,4 +119,26 @@ export default class IoSignService {
         }
       });
     });
+
+  /**
+   * Get the QTSP clauses
+   */
+  public readonly getQtspClausesMetadata = (): Promise<
+    IResponseErrorInternal | IResponseSuccessJson<QtspClausesMetadataDetailView>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getQtspClausesMetadata({});
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 500:
+            return ResponseErrorInternal(
+              readableProblem((response.value as unknown) as ProblemJson)
```

Comment:
can we decode this into `ProblemJson` instead of casting (with a` getOrElse()` fallback )?

Reply:
I added a decode with fallback. Thanks!

Reply:
remember to add the ToDO